### ISSUE TITLE
Test failures on GitTest and DateTest

### ISSUE THREAD

**Issue by AltairFromAquila** [cieslarmichal/faker-cxx#222]
There are a few issues regarding the tests:

 1. When I run the tests, I've got 9 failed ones:
```
[----------] Global test environment tear-down
[==========] 424 tests from 34 test suites ran. (7490 ms total)
[  PASSED  ] 415 tests.
[  FAILED  ] 9 tests, listed below:
[  FAILED  ] GitTest.shouldGenerateCommitDate
[  FAILED  ] GitTest.shouldGenerateCommitEntry
[  FAILED  ] DateTest.shouldGeneratePastDate
[  FAILED  ] DateTest.shouldGenerateRecentDate
[  FAILED  ] DateTest.shouldGenerateFutureDate
[  FAILED  ] DateTest.shouldGenerateSoonDate
[  FAILED  ] DateTest.shouldGenerateBirthDateByAge
[  FAILED  ] DateTest.shouldGenerateBirthDateByExactYear
[  FAILED  ] DateTest.shouldGenerateBirthDateByRangeYear

 9 FAILED TESTS
``` 
 2. There is a warning in the GitTest.cpp file:
  ```cpp
 unsigned long branchSplit = faker::StringHelper::split(branch, "-").size();
 ``` 
 3. The GitTest.cpp file isn't properly formatted.

**Comment by AltairFromAquila** [comment]
@cieslarmichal 
I found why I got those errors, and I made a change in the way time points are serialized. That change should add support to the MinGW GCC compiler (meaning that basically everyone who uses GCC on Windows should be able to run those tests without problems).

I ran the test locally compiled with the GCC and the MSVC compiler, and in both cases, it passed all the tests.

**Comment by cieslarmichal** [comment]
Wow, nice, good job! 😀 Willl you open a PR? I saw in changes you include fmt and use std::format, I think you should use fmt::format.

**Comment by AltairFromAquila** [comment]
I'm going to change std::format to fmt::format to make it more consistent with the rest of the codebase, and then I'll submit a PR.

**Comment by cieslarmichal** [comment]
Ok, thanks, the reason for fmt::format is apple clang++ does not support std::format, so I changed all std::format to fmt

**Comment by cieslarmichal** [comment]
@AltairFromAquila could you please submit a PR?

**Comment by AltairFromAquila** [comment]
@cieslarmichal

First of all, I apologize for the recent lack of updates about this issue.

Now, regarding a PR, I encountered a problem when changing std::format to fmt::format: When calling fmt::format, there is a line where two `std::time_t` values are subtracted and the result is stored in a `long`, but given that my compiler defines `std::time_t` as `long long` and that the tests are compiled using the `-Werror` flag, swapping std::format to fmt::format wasn't possible. I opened an [issue](https://github.com/fmtlib/fmt/issues/3717) about this on the fmt repo, and I was about to submit a PR there to close this issue, but, lately, I haven't been able to find time for that.

I can still submit a PR with the current changes, but it still wouldn't work on all compilers.
